frontend/src/metabase/lib/expressions/resolver.js

```diff
@@ -166,12 +166,13 @@ export function resolve(expression, type = "expression", fn = undefined) {
       return resolve(operand, args[i], fn);
     });
     return [op, ...resolvedOperands];
-  } else if (
-    !isCompatible(
-      type,
-      typeof expression === "boolean" ? "expression" : typeof expression,
-    )
-  ) {
+  }
+
+  if (type === "expression" && typeof expression === "boolean") {
+    return expression;
```

Comment:
This is rather a quick hack right now to let `F(X) < Y` marked as invalid (where F is a function returning boolean).
Expect a more comprehensive follow-up PR to tackle extensive comparison ops with booleans.